src/reynir/bintokenizer.py

```diff
@@ -1287,6 +1307,37 @@ def compatible(pn: PersonNameTuple, npn: PersonNameTuple) -> bool:
                 namespan = token.original or ""
                 patronym = False
                 while True:
+                    next_next_token = None
+                    # Deal with lowercase middle name abbreviations when auto uppercasing
+                    if auto_uppercase and next_token.txt in MIDDLE_NAME_ABBREVS:
+                        next_next_token = next(token_stream)
+                        following_tokens.append(next_next_token)
+
+                        # Check if next token is a period
+                        if next_next_token.punctuation == ".":
+                            # Join to middle name abbreviation
+                            next_token = next_token.concatenate(next_next_token)
+                            following_tokens.popleft()
+
+                            try:
+                                # Remove sentence end/start tokens
+                                while (
+                                    next_next_token.punctuation == "."
+                                    or next_next_token.kind in SENTENCE_END_START_TOKENS
+                                ):
+                                    next_next_token = next(token_stream)
```

Comment:
Shouldn't you be appending to following_tokens here, in this inner while loop?

Reply:
The reason I don't append to following_tokens is to remove the sentence delimiter tokens (`S_BEGIN`,`S_END`), which get added in the `annotate` phase before `parse_phrases_2` gets invoked.

I want to interpret e.g. the _k_ and _._ in _þorlákur k. sæmundarson_ as a single token without the `S_BEGIN` and `S_END` tokens. Maybe we want to keep the S_END token though if there are no tokens after the name?

Reply:
We want to keep S_END, yes, it may be relied upon in subsequent passes.

Reply:
Fixed!